WarrickSmith/WS-Portfolio-New
# Update grid

## Summary
This PR adds initial components and styling for a React portfolio project using TypeScript, Styled Components, framer-motion and Vite.

## Key Changes

- Adds package.json, tsconfig.json and vite config files to set up build tools and TypeScript
- Implements GlobalStyle component with root styles and CSS variables
- Creates MainPage component with grid layout and card interactions
- Implements WordSlider component for animated text transitions
- Sets up App component as root wrapper
- Adds index.html and entry point main.tsx

## Testing Instructions

### To test the changes:

- Run npm install
- Run npm run dev
- View app at localhost:3000
- Click cards to see interactions
- Verify styles are applied globally

## Issues
Resolves #1 - adds initial scaffolding and styling
